### ISSUE TITLE
PSI: please advice on unperled version

### ISSUE THREAD

**Issue by FlorianHeigl** [hweidner/check-mk-goodies#2]
Hi,

I have quite a zoo of systems, and can't assume perl being around.
(i.e. Alpine RISC-V with tiny ram/disk). I'm not yet sure how much it matters. 
But anyway - I wanna make an awk version.

Can you please share with me some example outputs? I'm enabling PSI on some systems already but wanna make sure I will end up with 100% identical output. (Did I mention I'm not a CS student or parser guru so my awk skills are not... great)

Would be nice if you can help me with a bit of regression testing food.

**Comment by FlorianHeigl** [comment]
ok i have it working on an Oracle Linux 8 now, Alpine I might still need to do some more things. I wonder - how is your experience with the check as it is - as it's a local check with less graphing, did you miss fancy or was it sufficient?
how about central config? did you ever need to change your warn/crit parameters?

do you have already found your flow with responding to it? do you go upgrade RAM where you might need to? Do you wait it out and observe for some time? Do you treat it as a status lamp or as an alert?

We can chat on some other channel if you want. I am probably gonna take this as tiny 'winter project' later in january.

**Comment by hweidner** [comment]
> I wonder - how is your experience with the check as it is - as it's a local check with less graphing, did you miss fancy or was it sufficient?

Yes, a "real" check plugin would allow smarter graphing and central configuration. I implemented it as a Local Check, because the plugin interface changed a lot over the time, while Local Checks remained stable since the beginning of Checkmk.

> how about central config? did you ever need to change your warn/crit parameters?

For me personally, this is not an issue. I use Puppet at work and Ansible at home/side projects for configuration management.

> do you have already found your flow with responding to it? do you go upgrade RAM where you might need to? Do you wait it out and observe for some time? Do you treat it as a status lamp or as an alert?

I use the PSI checks mainly for collecting historical performance data, not for alerting/notifications. Whenever an application works slower than expected, I dig into the performance data for a better understanding of the reasons. PSI values are a lot similar to the CPU load: the change over time is much more interesting than the concrete values themselves.

**Comment by hweidner** [comment]
I close this issue as there have been no further discussions within the last 1.5 years.

**Comment by FlorianHeigl** [comment]
Hey, just to say thanks. I'd worked on this back then and your input was useful.
But I have some health problems and couldn't follow up with a different implementation.
Have  a good time!
